File: 21.py
# -*- coding:utf-8 -*-
class Solution:
    def isOod(self,num:int)->bool:
        return not self.isEven(num)
    def isEven(self,num:int)->bool:
        return num % 2 == 0
    def reOrderArray(self, array:list)->list:
        OodList = []
        EvenList = []
        for num in array:
            if(self.isOod(num)):
                OodList.append(num)
            else:
                EvenList.append(num)
        OodList.extend(EvenList)
        return OodList
# write code here

File: 21.cpp
//
// Created by zb198 on 2019/10/6.
//
#include <vector>
using namespace std;
class Solution {
private:
    bool isOod(int num){
        return num % 2 != 0;
    }
public:
    void reOrderArray(vector<int> &array) {
        vector<int > OodList;
        vector<int > EvenList;
        for (int i=0;i<array.size();i++){
            if(isOod(array[i])){
                OodList.push_back(array[i]);
            } else{
                EvenList.push_back(array[i]);
            }
        }
        int j = 0;
        for(auto Iter = OodList.begin();Iter!=OodList.end();Iter++){
            array[j] = *Iter;
            j++;
        }
        for(auto Iter = EvenList.begin();Iter!=OodList.end();Iter++){
            array[j] = *Iter;
            j++;
        }

    }
};
